File: tournaments/buildPalindrome/buildPalindrome.cs
string buildPalindrome(string st) {
    if (null == st || 1 >= st.Length || isPalindrome(st)) {
        return st;
    }

    var n = st.Length;
    var sb = new StringBuilder(2 * n);
    sb.Append(st);

    char[] charArray = st.ToCharArray();
    Array.Reverse(charArray);
    sb.Append(charArray);
    var result = sb.ToString();

    for (int i = 0; n > i; ++i) {
        if (isPalindrome(sb.Remove(n, 1).ToString())) {
            result = sb.ToString();
        }
    }

    return result;
}

bool isPalindrome(string s) {
    if (null == s || 0 == s.Length) {
        return false;
    }

    if (1 == s.Length) {
        return true;
    }

    for (int i = 0; s.Length / 2 > i; ++i) {
        if (0 != s[i] - s[s.Length - 1 - i]) {
            return false;
        }
    }

    return true;
}

File: tournaments/buildPalindrome/buildPalindrome.cpp
std::string buildPalindrome(std::string st) {
  bool canConvert;
  for (int i = st.size(); ; i++) {
    canConvert = true;
    for (int j = 0; j < i - j - 1; j++) {
      if (i - j - 1 < st.size() && st[j] != st[i - j - 1]) {
        canConvert = false;
        break;
      }
    }
    if (canConvert) {
      for (int j = st.size(); j < i; j++) {
        st += st[i - j - 1];
      }
      return st;
    }
  }
}
